samples/sample-client-common/src/main/java/com/flipkart/aesop/sample/client/common/events/MysqlBinLogEventImpl.java

```diff
@@ -116,15 +126,79 @@ public DbusOpcode getEventType()
 		return this.eventType;
 	}
 
+	public Map<String, Object> getRowChangeMap() { return rowChangeMap; }
+
 	public String toString()
 	{
 		StringBuilder builder = new StringBuilder();
 		builder.append("MysqlBinLogEventImpl [");
 		builder.append("entityName : " + getEntityName() + ", ");
 		builder.append("eventType : " + getEventType().toString() + ", ");
 		builder.append("pKeyList : " + getPrimaryKeyList().toString() + ", ");
-		builder.append("keyValuePairs : " + getKeyValuePair().toString());
+		builder.append("keyValuePairs : " + getKeyValuePair().toString() + ", ");
+		builder.append("rowChangeMap : " + getRowChangeMap().toString());
 		return builder.toString();
 	}
 
+	/**
+	 * Check whether given field is rowChangeField
+	 * @param schema
+	 * @return name of rowChangeField
+	 */
+	private String getRowChangeField(Schema schema)
+	{
+		return (SchemaHelper.getMetaField(schema, META_ROW_CHANGE_FIELD));
+	}
+
+	/**
+	 * Generates a hash storing fieldName to intended MysqlDataType from schema
+	 * @param schema
+	 * @return Map containg FieldName to Mysql Type mapping
+	 */
+	private Map<String, String> fieldToDataTypeMap(Schema schema)
+	{
+		Map<String, String> map = new HashMap <String, String>(schema.getFields().size());
+		for (Schema.Field field : schema.getFields())
+		{
```

Comment:
This looks like a lot of duplicate code that is the same as what is present in AbstractEventFactory. Is there no way to refactor this out?